pkg/network/dns/stats.go

```diff
@@ -183,6 +187,34 @@ func (d *dnsStatKeeper) GetAndResetAllStats() StatsByKeyByNameByType {
 	return ret
 }
 
+func (d *dnsStatKeeper) WaitForDomain(domain string) error {
+	for {
+		select {
+		case <-time.After(waitForDomainTimeout):
+			return fmt.Errorf("domain %v did not appear within %v", domain, waitForDomainTimeout)
+		default:
+			if d.hasDomain(domain) {
+				return nil
+			}
+			time.Sleep(time.Millisecond * 10)
```

Comment:
You could use a ticker as another case for the select here.

Reply:
I'll change this to get the PR through. 

Is there a reason to prefer it one way or another? They're equivalent, and I feel that this version subjectively reads a little nicer.

Reply:
Its more go-like, plus the sleep would delay checking the deadline, while the ticker won't.